codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameCodec.java

```diff
@@ -216,7 +216,7 @@ public void onStreamClosed(Http2Stream stream) {
 
         @Override
         public void onGoAwayReceived(final int lastStreamId, long errorCode, ByteBuf debugData) {
-            ctx.fireChannelRead(new DefaultHttp2GoAwayFrame(lastStreamId, errorCode, debugData));
+            ctx.fireChannelRead(new DefaultHttp2GoAwayFrame(lastStreamId, errorCode, debugData.retain()));
```

Comment:
@normanmaurer Won't `fireChannelRead` occur immediately since we're in the event loop?

I'm not aware of this being an issue in gRPC and I'm concerned that doing this may introduce a leak there.  @ejona86 WDYT?